presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -102,7 +104,7 @@ public NativeExecutionTask(
         this.taskResultFetcher = new HttpNativeExecutionTaskResultFetcher(
                 updateScheduledExecutor,
                 this.workerClient,
-                Optional.empty());
+                Optional.of(maxRequestTimeout));
```

Comment:
It turns out you are passing this timeout to result fetcher. Then it should be called resultFetchingRequestTimeout.

I think it will be also nice to add timeout to task info fetching. Currently I'm thinking the taskinfo could be stale if requests keep failing. And there will be no signals to tell that scenario. We might need some mechanism to address that.

Reply:
Renamed the config to `resultFetchingRequestTimeout`. Per our offline discussion, I'll add the changes to taskInfo fetching in a followup PR.